plugins/modules/mso_remote_location.py

```diff
@@ -0,0 +1,249 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2022, Akini Ross (@akinross) <akinross@cisco.com>
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = r'''
+---
+module: mso_remote_location
+short_description: Manages remote locations
+description:
+- Manage remote locations on Cisco ACI Multi-Site.
+author:
+- Akini Ross (@akinross)
+options:
+  remote_location:
+    description:
+    - The remote location's name.
+    type: str
+  remote_description:
+    description:
+    - The remote location's description.
+    type: str
+  remote_protocol:
+    description:
+    - The protocol used to export to the remote server.
+    - If the remote location is a Windows server, you must use the C(sftp) protocol.
+    choices: [ scp, sftp ]
+    type: str
+  remote_host:
+    description:
+    - The host name or IP address of the remote server.
+    type: str
+  remote_path:
+    description:
+    - The full path to a directory on the remote server where backups are saved.
+    - The path must start with a slash (/) character and must not contain periods (.) or backslashes (\).
+    - The directory must already exist on the server.
+    type: str
+  remote_port:
+    description:
+    - The port used to connect to the remote server.
+    default: 22
+    type: int
+  remote_authentication_type:
+    description:
+    - The authentication method used to connect to the remote server.
+    choices: [ password, ssh ]
+    type: str
+  remote_username:
+    description:
+    - The username used to log in to the remote server.
+    type: str
+  remote_password:
+    description:
+    - The password used to log in to the remote server.
+    type: str
+  remote_ssh_key:
+    description:
+    - The ssh key used to log in to the remote server.
+    - The ssh key must be provided in PEM format.
+    - The ssh key must be a single line string with linebreaks represent as "\n".
+    type: str
+  remote_ssh_passphrase:
+    description:
+    - The passphrase used to log in to the remote server.
```

Comment:
The private ssh key passphrase to use to login to the remote server.

Reply:
ok